xbmc/input/ButtonTranslator.cpp

```diff
@@ -125,18 +125,18 @@ bool CButtonTranslator::Load()
 
 bool CButtonTranslator::LoadKeymap(const std::string& keymapPath)
 {
-  CXBMCTinyXML xmlDoc;
+  CXBMCTinyXML2 xmlDoc;
 
   CLog::Log(LOGINFO, "Loading {}", keymapPath);
   if (!xmlDoc.LoadFile(keymapPath))
   {
-    CLog::Log(LOGERROR, "Error loading keymap: {}, Line {}\n{}", keymapPath, xmlDoc.ErrorRow(),
-              xmlDoc.ErrorDesc());
+    CLog::Log(LOGERROR, "Error loading keymap: {}, Line {}\n{}", keymapPath, xmlDoc.ErrorLineNum(),
+              xmlDoc.ErrorStr());
     return false;
   }
 
-  TiXmlElement* pRoot = xmlDoc.RootElement();
-  if (pRoot == nullptr)
+  auto* pRoot = xmlDoc.RootElement();
+  if (!pRoot)
```

Comment:
Any reason for the change here? The explicit checks are fine (this change is in a few places).

Reply:
Honestly input has been one of the last things i looked at regarding tinyxml2 usage. I started changing to the bool check because theres every way under the sun used everywhere else, so i just got into the rhythm of changing the checks to bool check for consistency (less typing than changing everything to == nullptr)

I dont really care which way, but id like some consistency, so tell me what is preferred

Reply:
I'd prefer avoiding implicit casting. `!` implicitly converts pointer to bool, but `!= nullptr` takes two pointers and returns a bool.

Similar to how I'd avoid writing `float x = 0.0;` and instead write `float x = 0.0f;` to maintain type consistency even though it's the same to the compiler.